tensorflow_probability/python/experimental/mcmc/streaming_reductions.py

```diff
@@ -0,0 +1,104 @@
+# Copyright 2018 The TensorFlow Probability Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ============================================================================
+"""Drivers for streaming reductions framework."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+# Dependency imports
+import warnings
+
+import tensorflow.compat.v2 as tf
+from tensorflow_probability.python.internal import samplers
+from tensorflow_probability.python.mcmc.internal import util as mcmc_util
+
+
+__all__ = [
+    'step_kernel',
+]
+
+
+def step_kernel(
+    num_steps,
+    current_state,
+    previous_kernel_results=None,
+    kernel=None,
+    return_final_kernel_results=False,
+    parallel_iterations=10,
+    seed=None,
+    name=None,
+):
+  """Takes `num_steps` repeated `TransitionKernel` steps from a supplied
+  `current_state`.
+
+  Args:
+    num_steps: Integer number of Markov chain steps.
+    current_state: `Tensor` or Python `list` of `Tensor`s representing the
+      current state(s) of the Markov chain(s).
+    previous_kernel_results: A `Tensor` or a nested collection of `Tensor`s
+      representing internal calculations made within the previous call to this
+      function (or as returned by the kernel's `bootstrap_results` if None).
+      Default value: `None` (i.e., return of kernel.bootstrap_results)
+    kernel: An instance of `tfp.mcmc.TransitionKernel` which implements one step
+      of the Markov chain.
+    return_final_kernel_results: If `True`, then the final kernel results are
+      returned alongside the chain state after `num_steps` steps are taken.
+    parallel_iterations: The number of iterations allowed to run in parallel. It
+      must be a positive integer. See `tf.while_loop` for more details.
+    seed: Optional, a seed for reproducible sampling.
+    name: Python `str` name prefixed to Ops created by this function.
+      Default value: `None` (i.e., 'mcmc_sample_chain').
+
+  Returns:
+    next_state, final_kernel_results: If `return_final_kernel_results` is
+      `True`, the kernel results after `num_steps` steps will be returned
+      alongside the resultant Markov chain state.
+    next_state: Conversely, if `return_final_kernel_results` is `False`,
+      only the resultant state will be returned.
```

Comment:
The convention for these is to list each member of the tuple on its own line, describing just those.  In this case, that would be

```
Returns:
  final_state: ...
  final_kernel_results: ...  Only returned if `return_final_kernel_results` was `True`.
```